docs/dev/table/sql.md

```diff
@@ -813,6 +813,222 @@ LIMIT 3
 
 {% top %}
 
+### Top-N
+
+TopN is used to calculate the maximum/minimum N records in a stream. It can be flexibly completed based on OVER window aggregation. The grammar is shown as below:
+
+Top-N queries ask for the N smallest or largest values ordered by columns. Both smallest and largest values sets are considered Top-N queries. Top-N queries are useful in cases where the need is to display only the N bottom-most or the N top-
+most records from batch/stream table on a condition. This result set can be used for further analysis.
+
+Flink uses the combination of a OVER window clause and a filter condition to express a Top-N query. With the power of OVER window `PARTITION BY` clause, Flink also supports per group Top-N. For example, the top five products per category that have the maximum sales in realtime. Top-N queries are supported for SQL on batch and streaming tables.
```

Comment:
some places is batch/stream, some others are streaming? maybe we just use stream?

Reply:
I think we also support this in batch? 
The attention part and optimization part is only relative to streaming. I can explain it more detailed.